orchestrator/careplancontributor/handle_taskfiller_test.go

```diff
@@ -3,171 +3,257 @@ package careplancontributor
+		},
+		{
+			name: "error: primary task, invalid (missing SCP profile)",
+			notificationTask: deep.AlterCopy(primaryTask, func(t *fhir.Task) {
+				t.Meta.Profile = []string{"SomeOtherProfile"}
+			}),
+		},
+		{
+			name: "error: primary task, without Requester",
+			notificationTask: deep.AlterCopy(primaryTask, func(t *fhir.Task) {
+				t.Requester = nil
+			}),
+			expectedError: errors.New("task is not valid - skipping: validation errors: Task.requester is required but not provided"),
+		},
+		{
+			name: "error: primary task, without Owner",
+			notificationTask: deep.AlterCopy(primaryTask, func(t *fhir.Task) {
+				t.Owner = nil
+			}),
+			expectedError: errors.New("task is not valid - skipping: validation errors: Task.owner is required but not provided"),
+		},
+		{
+			name: "primary task, status=received, should not process",
+			notificationTask: deep.AlterCopy(primaryTask, func(t *fhir.Task) {
+				t.Status = fhir.TaskStatusReceived
+			}),
+		},
+		{
+			name: "primary task, status=accepted, should not process",
+			notificationTask: deep.AlterCopy(primaryTask, func(t *fhir.Task) {
+				t.Status = fhir.TaskStatusAccepted
+			}),
+		},
+		{
+			name: "primary task, status=accepted, should not process",
+			notificationTask: deep.AlterCopy(primaryTask, func(t *fhir.Task) {
+				t.Status = fhir.TaskStatusRejected
+			}),
+		},
+		{
+			name: "primary task, status=on-hold, should not process",
+			notificationTask: deep.AlterCopy(primaryTask, func(t *fhir.Task) {
+				t.Status = fhir.TaskStatusOnHold
+			}),
+		},
+		{
+			name: "primary task, status=cancelled, should not process",
+			notificationTask: deep.AlterCopy(primaryTask, func(t *fhir.Task) {
+				t.Status = fhir.TaskStatusCancelled
+			}),
+		},
+		{
+			name: "primary task, status=completed, should not process",
+			notificationTask: deep.AlterCopy(primaryTask, func(t *fhir.Task) {
+				t.Status = fhir.TaskStatusCompleted
+			}),
+		},
+		{
+			name: "primary task, status=failed, should not process",
+			notificationTask: deep.AlterCopy(primaryTask, func(t *fhir.Task) {
+				t.Status = fhir.TaskStatusFailed
+			}),
+		},
+		{
+			name: "primary task, status=ready, should not process",
+			notificationTask: deep.AlterCopy(primaryTask, func(t *fhir.Task) {
+				t.Status = fhir.TaskStatusReady
+			}),
+		},
+		{
+			name: "primary task, status=in-progress, should not process",
+			notificationTask: deep.AlterCopy(primaryTask, func(t *fhir.Task) {
+				t.Status = fhir.TaskStatusInProgress
+			}),
+		},
+		{
+			name: "error: primary task, unknown service is requested (primary Task.focus(ServiceRequest).code is not supported)",
+			serviceRequest: deep.AlterCopy(serviceRequest, func(sr *fhir.ServiceRequest) {
+				sr.Code.Coding[0].Code = to.Ptr("UnknownServiceCode")
+			}),
+			expectedError: errors.New("failed to process new primary Task: ServiceRequest.code does not match any offered services"),
```

Comment:
I think we're missing a test where the same check is done for the "Task.reasonCode or Task.reasonReference does not match any workflow for service"